cmd/datamon/cmd/cli_fuse_test.go

```diff
@@ -3,166 +3,331 @@
 package cmd
 
 import (
+	"bufio"
+	"io"
 	"io/ioutil"
+	"math/rand"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"regexp"
+	"strconv"
 	"strings"
+	"sync"
 	"testing"
 	"time"
 
+	"github.com/oneconcern/datamon/pkg/dlogger"
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
 )
 
-func testBundleMount(t *testing.T, testType string) {
+var (
+	buildBinaryonce sync.Once
+	testBinary      string
+)
+
+func testLogger(t testing.TB) *zap.Logger {
+	// a logger for the test sequence itself
+	return dlogger.MustGetLogger("debug").With(zap.String("test", t.Name()))
+}
+func testCommand(t testing.TB, target string, args ...string) (*exec.Cmd, io.ReadCloser) {
+	// NOTE: this starts a process, not a goroutine like for other tests
+	// That is primarily because we want to send a signal to the process
+	// in order to unmount.
+	cmd := exec.Command(target, args...)
+	cmd.Env = append(cmd.Env, "DATAMON_CONTEXT="+os.Getenv("DATAMON_CONTEXT")) // TODO(fred): not sure this is used
+	cmd.Env = append(cmd.Env, "DATAMON_GLOBAL_CONFIG="+os.Getenv("DATAMON_GLOBAL_CONFIG"))
+	cmd.Env = append(cmd.Env, "HOME="+os.Getenv("HOME"))
+	cmd.Env = append(cmd.Env, "PATH="+os.Getenv("PATH"))
+	cmd.Env = append(cmd.Env, "GOOGLE_APPLICATION_CREDENTIALS="+os.Getenv("GOOGLE_APPLICATION_CREDENTIALS"))
+	pipeOut, err := cmd.StdoutPipe()
+	require.NoError(t, err)
+	pipeErr, err := cmd.StderrPipe()
+	require.NoError(t, err)
+
+	// combine stdout & stderr, tee this output to os.Stdout and return pipe reader for output scanning
```

Comment:
Since we want to test if the output of a cli command is correctly being routed to STDOUT, we might want to keep the separation of what gets logged to which channel. Since operations such a list needs to be piped in shell to other unix tools, we have to add a test to make sure the output is to STD OUT.

Reply:
ok will remove the combined output

Reply:
@kerneltime after some more inspection, it so happens that I am asserting on both output... 
I am deferring the more accurate separate assertions on stderr and stdout as a TODO (see #374).
Deferred.